packages/swapper/src/swappers/ChainflipSwapper/getChainflipTradeQuote/getTradeQuote.ts

```diff
@@ -0,0 +1,276 @@
+    `${brokerUrl}/quotes-native?apiKey=${apiKey}&sourceAsset=${sellChainflipChainKey}&destinationAsset=${buyChainflipChainKey}&amount=${sellAmount}&commissionBps=${serviceCommission}`,
+  );
+
+  // TODO: Throw SellAmountBelowMinimum if that's the cause (parse error response)
+  if (maybeQuoteResponse.isErr()) {
+    return Err(
+      makeSwapErrorRight({
+        message: 'Quote request failed',
+        code: TradeQuoteError.NoRouteFound,
+      }),
+    )
+  }
+
+  const { data: quoteResponse } = maybeQuoteResponse.unwrap()
+  
+  const getGasFee = async () => {
+    const { chainNamespace } = fromAssetId(sellAsset.assetId);
+
+    switch (chainNamespace) {
+      case CHAIN_NAMESPACE.Evm: {
+        const sellAdapter = deps.assertGetEvmChainAdapter(sellAsset.chainId);
+        return await getEvmTxFees({
+          adapter: sellAdapter,
+          supportsEIP1559: (input as GetEvmTradeQuoteInput).supportsEIP1559,
+        });
+      }
+
+      case CHAIN_NAMESPACE.Utxo: {
+        // TODO: Figure out BTC gas calc
+        return undefined;
+        // const sellAdapter = deps.assertGetUtxoChainAdapter(sellAsset.chainId)
+        // return await getUtxoTxFees({
+        //   sellAmountCryptoBaseUnit: sellAmount,
+        //   vault,
+        //   opReturnData,
+        //   pubkey,
+        //   sellAdapter,
+        // })
+      }
+      
+      case CHAIN_NAMESPACE.Solana: {
+        // TODO: Solana gas calc
+        return undefined;
+      }
+    }
+    
+    return undefined;
+  }
+  
+  const getFeeAsset = (fee: ChainflipBaasQuoteQuoteFee) => {
+    if (fee.type === "ingress" || fee.type === "boost")
+      return sellAsset;
+
+    if (fee.type === "egress")
+      return buyAsset;
+
+    if (fee.type === "liquidity" && fee.asset == sellChainflipChainKey)
+      return sellAsset;
+
+    if (fee.type === "liquidity" && fee.asset == buyChainflipChainKey)
+      return buyAsset;
+
+    if (fee.type === "liquidity" && fee.asset == "usdc.eth")
+      return usdcAsset;
+
+    if (fee.type === "network")
+      return usdcAsset;
+  }
+  
+  const getProtocolFees = (singleQuoteResponse: ChainflipBaasQuoteQuote) => {
+    const protocolFees: Record<AssetId, ProtocolFee> = {}
+
+    for (const fee of singleQuoteResponse.includedFees!) {
+      if (fee.type === "broker")
+        continue;
+
+      const asset = getFeeAsset(fee)!
+      if (!(asset.assetId in protocolFees)) {
+        protocolFees[asset.assetId] = {
+          amountCryptoBaseUnit: "0",
+          requiresBalance: false,
+          asset: asset,
+        }
+      }
+ 
+      protocolFees[asset.assetId].amountCryptoBaseUnit =
+        (BigInt(protocolFees[asset.assetId].amountCryptoBaseUnit) + BigInt(fee.amountNative!)).toString();
+    }
+
+    return protocolFees;
+  }
+  
+  const quotes = [];
+  
+  for (const singleQuoteResponse of quoteResponse) {
+    const isStreaming = singleQuoteResponse.type === "dca";
```

Comment:
🤯 TIL, that's amazing! https://blog.chainflip.io/how-dca-swaps-work-on-chainflip/
Will be a bit more cumbersome to integrate in the app as Chainflip is going to be the first non-THOR swapper supporting streaming swaps, so we may want to disable these as out-of-scope for initial launch if that requires too much effort to generalize and make it work for Chainflip

Reply:
I can put it behind a feature toggle for now? No idea (yet) where streaming gets reflected, I guess in the status after you send the swap?

Reply:
Behind `REACT_APP_FEATURE_CHAINFLIP_DCA` for now